runner/src/runner.rs

```diff
@@ -0,0 +1,629 @@
+use crate::error::Error;
+use bytes::Bytes;
+use codec::Decode;
+use jsonrpsee::{
+    core::client::{Client as WsClient, ClientT},
+    rpc_params,
+    ws_client::WsClientBuilder,
+};
+use mockall_double::double;
```

Comment:
Gate this with `#[cfg(test)]` and then move the dependencies to dev.

Reply:
`mockall` isn't meant to be used with the `#[cfg(test)]` macro (it should just work, declared as dev-dependencies). `mockall_double` has to be added as a build dependency with the mention that it's very small. The problem is that their [example](https://docs.rs/mockall/latest/mockall/index.html#mocking-structs) to mock structs fails unless `mockall` is also a build dependency.